### ISSUE TITLE
v3 'could not find a repo containing the given URL'

### ISSUE THREAD

**Issue by OJFord** [helm/helm#5981]
```sh
$ helm version
version.BuildInfo{Version:"v3.0.0-alpha.1", GitCommit:"b9a54967f838723fe241172a6b94d18caf8bcdca", GitTreeState:"clean"}
```

Reproduce:
```yaml
# requirements.yaml
dependencies:
  - name: loki
    version: 0.10.0
    repository: https://grafana.github.io/loki/charts
```
```sh
$ helm init
Creating /tmp/helm3/plugins
Creating /tmp/helm3/starters
Creating /tmp/helm3/repository/repositories.yaml
Adding stable repo with URL: https://kubernetes-charts.storage.googleapis.com
$HELM_HOME has been configured at /tmp/helm3/.
Happy Helming!
$
$ helm repo add loki https://grafana.github.io/loki/charts
"loki" has been added to your repositories
$
$ helm dependency update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "loki" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Downloading loki from repo https://grafana.github.io/loki/charts
Save error occurred:  could not download https://grafana.github.io/loki/charts/loki-0.10.0.tgz: could not find a repo containing the given URL
Deleting newly downloaded charts, restoring pre-update state
Error: could not download https://grafana.github.io/loki/charts/loki-0.10.0.tgz: could not find a repo containing the given URL
```

**Comment by karuppiah7890** [comment]
This is a duplicate of https://github.com/helm/helm/issues/5936 and it's fixed in this b9878bd912ffc18a9addd373ec8c78f81f6baf84 commit by this https://github.com/helm/helm/pull/5940 PR. 

To get the latest binary with the fix, you can clone the repo at `$GOPATH/src/helm.sh/helm`, checkout `dev-v3` branch and build it. I can't think of any other way

**Comment by OJFord** [comment]
Ah, sorry I missed that one.

I worked around it by downloading manually, that's good enough for me until next release.

Cheers.
